spec/controllers/answers_controller_spec.rb

```diff
@@ -1,137 +1,215 @@
+          expect(response).to render_template(:mark_as_the_best)
+        end
       end
     end
-    
-    context 'with invalid attributes' do
-      it 'does not save the answer' do
-        expect { post :create, params: { question_id: question, answer: attributes_for(:answer, :invalid) } }.to_not change(question.answers, :count)
-      end
 
-      it 're-render new view' do
-        post :create, params: { question_id: question, answer: attributes_for(:answer, :invalid) }
-        expect(response).to render_template(:new)
+    context "when it is not an author" do
+      before { login(users[1]) }
+      
+      describe 'PATCH #update' do
+        context 'with valid attributes' do
+          it 'not assigns the requested answer to @answer' do
+            patch :update, params: { id: answer, answer: attributes_for(:answer), format: :js }
+            expect(assigns(:answer)).not_to eq(answer)
+          end
+
+          it 'not changes answer attributes' do
+            patch :update, params: { id: answer, answer: { body: 'new body' }, format: :js } 
+            answer.reload
+
+            expect(answer.body).to eq('MyText')
+          end
+
+          it 'not renders update view' do
+            patch :update, params: { id: answer, answer: attributes_for(:answer), format: :js }
+            expect(response).to render_template :update 
+          end
+        end
+
+        context 'with invalid attributes' do
+          before { patch :update, params: { id: answer, answer: attributes_for(:answer, :invalid), format: :js } }
+
+          it 'does not change answer' do
+            answer.reload
+
+            expect(answer.body).to eq('MyText')
+          end
+
+          it 'renders update view' do
+            expect(response).to render_template(:update)
+          end
+        end
       end
-    end
-  end
 
-  describe 'PATCH #update' do
-    before { login(user) }
+      describe 'DELETE #destroy' do
+        it 'not deletes the answer' do
+          answer
+          expect { delete :destroy, params: { id: answer }, format: :js }.not_to change(Answer, :count)
+        end
 
-    context 'with valid attributes' do
-      it 'assigns the requested answer to @answer' do
-        patch :update, params: { id: answer, answer: attributes_for(:answer) } 
-        expect(assigns(:answer)).to eq(answer)
+        it 'renders destroy view' do
+          delete :destroy, params: { id: answer }, format: :js 
+          expect(response).to render_template(:destroy) 
+        end
       end
 
-      it 'changes answer attributes' do
-        patch :update, params: { id: answer, answer: { body: 'new body' } } 
-        answer.reload
-
-        expect(answer.body).to eq('new body')
+      describe 'PATCH #mark_as_the_best' do
+        it 'changes answer attribute' do
+          patch :mark_as_the_best, params: { id: answer, format: :js } 
+          answer.reload
+          
+          expect(answer.is_best).to eq(0)
```

Comment:
https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/predicate-matchers